unit-tests-java8/src/test/java/org/eclipse/collections/test/list/UnmodifiableListTestCase.java

```diff
@@ -10,34 +10,21 @@
 
 package org.eclipse.collections.test.list;
 
-import java.util.List;
-
-import org.eclipse.collections.test.UnmodifiableCollectionTestCase;
 import org.junit.Test;
 
-public interface UnmodifiableListTestCase extends UnmodifiableCollectionTestCase, ListTestCase
+import static org.junit.Assert.assertThrows;
+
+public interface UnmodifiableListTestCase extends FixedSizeListTestCase
 {
     @Override
-    @Test(expected = UnsupportedOperationException.class)
+    @Test
     default void List_set()
     {
-        List<Integer> list = this.newWith(1, 2, 3);
-        list.set(1, 4);
-    }
+        assertThrows(UnsupportedOperationException.class, () -> this.newWith(1).set(0, 4));
+        assertThrows(UnsupportedOperationException.class, () -> this.newWith(1, 2).set(0, 4));
+        assertThrows(UnsupportedOperationException.class, () -> this.newWith(1, 2, 3).set(0, 4));
 
-    @Override
-    @Test(expected = UnsupportedOperationException.class)
-    default void List_set_negative()
-    {
-        List<Integer> list = this.newWith(1, 2, 3);
-        list.set(-1, 4);
-    }
-
-    @Override
-    @Test(expected = UnsupportedOperationException.class)
-    default void List_set_out_of_bounds()
-    {
-        List<Integer> list = this.newWith(1, 2, 3);
-        list.set(4, 4);
+        assertThrows(UnsupportedOperationException.class, () -> this.newWith().set(-1, 4));
+        assertThrows(UnsupportedOperationException.class, () -> this.newWith(1, 2, 3).set(4, 4));
```

Comment:
Combined the tests here too. Collections that are unmodifiable tend to throw UnsupportedOperationException no matter what, even if the method would have failed for other reasons like index out of bounds, and even if the method wouldn't have performed a mutation.